### PR TITLE
Get virtwho configure status by icon color

### PR DESCRIPTION
**Pull request by eko999**
After deployed the virtwho configure, we need to validate the configure status, but it's difficult to get the status by Text:
<span class="virt-who-config-report-status pficon-ok status-ok" title="" data-original-title="The virt-who report arrived within the interval"></span>

This commit will support to get the status from icon color:
`rgba(63, 156, 53, 1)': 'green'` => configure is deployed successfully.
`rgba(54, 54, 54, 1)': 'gray'`  => configure is not reported yet.

### CODE REVIEW

**Comment by eko999** [comment]
Test result is passed:
```
# pytest test_ui_virtwho.py  
2019-07-16 13:58:40 - conftest - DEBUG - Registering custom pytest_configure

============================================================================ test session starts ============================================================================
platform linux -- Python 3.7.3, pytest-4.6.3, py-1.7.0, pluggy-0.12.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /root/eko/robottelo, inifile: tox.ini
plugins: mock-1.10.4, xdist-1.29.0, cov-2.7.1, forked-1.0.2, services-1.3.1, flake8-1.0.1
collecting ... 2019-07-16 13:58:40 - conftest - DEBUG - BZ deselect is disabled in settings

collected 2 items                                                                                                                                                           

test_ui_virtwho.py .s                                                                                                                                                 [100%]

=================================================================== 1 passed, 1 skipped in 511.33 seconds ===================================================================

```

File: airgun/views/virtwho_configure.py [hunk 1]
```diff
@@ -19,13 +21,52 @@
...
+
+    def __init__(self, parent, locator=None, logger=None):
+        """Provide default locator value if it wasn't passed"""
+        if not locator:
+            locator = ".//span[contains(@class, 'virt-who-config-report-status')]"
+        Widget.__init__(self, parent, logger=logger)
+        self.locator = locator
+
+    @property
+    def color(self):
```
Comment (mirzal) [inline airgun/views/virtwho_configure.py]: I don't really like us obtaining generated color and doing decisions based on that. We should focus on message that this color carry.

Looking at [plugin code](https://github.com/theforeman/foreman_virt_who_configure/blob/master/app/helpers/foreman_virt_who_configure/configs_helper.rb#L44), I can see they add specific class based on information they want to provide. Can we rewrite that to provide `status` (not `color`) property and do decisions based on element classes?

File: airgun/views/virtwho_configure.py [hunk 2]
```diff
@@ -19,13 +21,52 @@
...
+class StatusIcon(GenericLocatorWidget):
+    """The status for virtwho configure can be: green, gray.
+
+    green: The virt-who report has not arrived within the interval,
+         which indicates there was no change on hypervisor
+    gray: The configuration was not deployed yet or the virt-who was
+        unable to report the status
+    """
+
+    def __init__(self, parent, locator=None, logger=None):
```
Comment (mirzal) [inline airgun/views/virtwho_configure.py]: Do you need this `__init__` at all? Can't you just define your icon locator as class variable? Look at [`RadioGroup` in `airgun/widgets.py`](https://github.com/SatelliteQE/airgun/blob/master/airgun/widgets.py#L66) (or pretty much anything else inheriting from `GenericLocatorWidget` in that file).

File: airgun/views/virtwho_configure.py [hunk 3]
```diff
@@ -19,13 +21,52 @@
...
+        }
+        return colors.get(
+            self.browser.element(self, parent=self.parent).value_of_css_property('color'),
+            'unknown'
+        )
+
+    def read(self):
+        """Returns current icon color"""
+        return self.color
+
```
Comment (mirzal) [inline airgun/views/virtwho_configure.py]: According to [widgetastic guidelines](https://widgetasticcore.readthedocs.io/en/latest/guidelines.html), all widgets should provide both `read()` and `fill()` methods. If `fill()` doesn't make sense for widget (as in your case), it should raise `ReadOnlyWidgetError` exception (see e.g. [`ItemsListReadOnly` in `airgun/widgets.py`](https://github.com/SatelliteQE/airgun/blob/master/airgun/widgets.py#L430)).

**Comment by eko999** [comment]
@mirzal , I changed the class and get the configure status by the element's attributes, thanks for your suggestion and please review again.

**Comment by eko999** [comment]
@mirzal , fill() method is added, and a new class `VirtwhoConfigureScript` is including, I need to use the class to get the code from  `<pre> code </pre>`.

**Comment by mirzal** [comment]
@omkarkhatavkar @vijay8451 @lhellebr Hey, this has been opened 2 weeks ago, and had last activity almost a week ago. Could you take a look and ACK, if you are ok with these changes? Then you can merge, please use "Squash and merge" (as there are a lot of commits).

**Pull request merged**